gitea-mcp-server: init at 0.2.0 and add conneroisu as maintainer

connerohnesorge opened the pull request:
This pull request introduces a new package for the Gitea MCP Server and adds a new maintainer to the `maintainers` list. Below are the key changes grouped by theme:

### Maintainer Updates:
* Added `connerohnesorge` as a new maintainer with the associated email, GitHub username, GitHub ID, and name in `maintainers/maintainer-list.nix`.

### Package Addition:
* Introduced a new package `gitea-mcp-server` in `pkgs/by-name/gi/gitea-mcp-server/package.nix`. This package is built using `buildGoModule`, fetches its source from Gitea, and includes metadata such as description, homepage, license, and maintainers. The package is designed to provide a Model Context Protocol (MCP) server for seamless integration with Gitea APIs.

[https://gitea.com/gitea/gitea-mcp](https://gitea.com/gitea/gitea-mcp)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

pilz0 commented on pkgs/by-name/gi/gitea-mcp-server/package.nix:
```diff
@@ -0,0 +1,44 @@
+  buildGoModule,
+  fetchFromGitea,
+}:
+buildGoModule rec {
```
rec to point to versions is deprecated, you shouldn't use it here.
```suggestion
buildGoModule (finalAttrs: {
```

pilz0 commented on pkgs/by-name/gi/gitea-mcp-server/package.nix:
```diff
@@ -0,0 +1,44 @@
+    mainProgram = "gitea-mcp";
+    maintainers = with lib.maintainers; [ connerohnesorge ];
+  };
+}
```
```suggestion
})
```

pilz0 commented on pkgs/by-name/gi/gitea-mcp-server/package.nix:
```diff
@@ -0,0 +1,44 @@
+  postInstall = ''
+    install -Dm644 README.md LICENSE -t $out/share/gitea-mcp-server
+  '';
+
```
Other MCP Servers have line too, so it prob needs permission for the local networking too. (https://github.com/NixOS/nixpkgs/blob/67d6ef8521f52bf9f0d479b1bceb660faae79639/pkgs/by-name/gi/github-mcp-server/package.nix#L29)
```suggestion
  __darwinAllowLocalNetworking = true;
```

pilz0 commented on pkgs/by-name/gi/gitea-mcp-server/package.nix:
```diff
@@ -0,0 +1,47 @@
+in
+buildGoModule (finalAttrs: {
+  pname = "gitea-mcp-server";
+  inherit (versionJSON) version;
```
```suggestion
  version = "0.2.0";
```

pilz0 commented on pkgs/by-name/gi/gitea-mcp-server/package.nix:
```diff
@@ -0,0 +1,47 @@
+    domain = "gitea.com";
+    owner = "gitea";
+    repo = "gitea-mcp";
+    tag = "v${versionJSON.version}";
```
```suggestion
    tag = "v${finalAttrs.version}";
```

connerohnesorge committed:
maintainers: add conneroisu

connerohnesorge committed:
gitea-mcp-server: init at 0.2.0

A typescript Model Context Protocol (MCP) server that integrates with
Gitea to allow retrieval of repository and organizational data.

pilz0 commented:
You will still need to squash your commits, but otherwise the PR looks good to me :)

pbsds commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414358`
Commit: `a36484e2979026d983a313ed9567ee12f551f854`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gitea-mcp-server</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gitea-mcp-server</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gitea-mcp-server</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gitea-mcp-server</li>
  </ul>
</details>
